rllib/algorithms/ppo/ppo.py

```diff
@@ -384,24 +414,31 @@ def training_step(self) -> ResultDict:
 
         # Update weights - after learning on the local worker - on all remote
         # workers.
-        if self.workers.num_remote_workers() > 0:
-            with self._timers[SYNCH_WORKER_WEIGHTS_TIMER]:
+        with self._timers[SYNCH_WORKER_WEIGHTS_TIMER]:
+            if self.workers.num_remote_workers() > 0:
                 from_worker = None
                 if self.config._enable_rl_trainer_api:
+                    # sync weights from trainer_runner to all rollout workers
                     from_worker = self.trainer_runner
                 self.workers.sync_weights(
                     from_worker=from_worker,
-                    policies=list(train_results.keys()),
+                    policies=policies_to_update,
                     global_vars=global_vars,
                 )
+            elif self.config._enable_rl_trainer_api:
+                weights = self.trainer_runner.get_weights()
+                self.workers.local_worker().set_weights(weights)
 
         if self.config._enable_rl_trainer_api:
             kl_dict = {
-                pid: pinfo[LEARNER_STATS_KEY].get("kl")
-                for pid, pinfo in train_results.items()
+                pid: train_results["loss"][pid].get("mean_kl_loss")
```

Comment:
What's the case where this KL key is NOT present in the ["loss"][pid] sub-dict? Do we need the `get` here?

Reply:
captured it in a TODO